src/components/popups.vue

```diff
@@ -212,152 +244,130 @@ import { Apis } from "bitsharesjs-ws";
+                this.specifics = mappedData.description;
+                this.incoming.vote_id = mappedData.vote_id;
 
-                let objextradata;
-                switch (entity_id[1]) {
-                case "5":
-                    entity = "committee member";
-                    objextradata = await Apis.instance()
-                        .db_api()
-                        .exec("get_objects", [[objdata[0].committee_member_account]]);
-                    this.specifics =
-                        "Commitee member: " +
-                        objextradata[0].name +
-                        "\nCommittee Member ID: " +
-                        this.incoming.params.id;
-                    this.incoming.vote_id = objdata[0].vote_id;
-                    break;
-                case "6":
-                    entity = "witness";
-                    objextradata = await Apis.instance()
-                        .db_api()
-                        .exec("get_objects", [[objdata[0].witness_account]]);
-                    this.specifics =
-                        "Witness: " +
-                        objextradata[0].name +
-                        "\nWitness ID: " +
-                        this.incoming.params.id;
-                    this.incoming.vote_id = objdata[0].vote_id;
-                    break;
-                case "14":
-                    entity = "worker proposal";
-                    objextradata = await Apis.instance()
-                        .db_api()
-                        .exec("get_objects", [[objdata[0].worker_account]]);
-                    this.specifics =
-                        "Proposal: " +
-                        objdata[0].name +
-                        "\nProposal ID: " +
-                        this.incoming.params.id +
-                        "\nDaily Pay: " +
-                        this.formatMoney(objdata[0].daily_pay / Math.pow(10, 5), 5) +
-                        "BTS\nWorker: " +
-                        objextradata[0].name;
-                    this.incoming.vote_id = objdata[0].vote_for;
-                    break;
-                }
-                this.genericmsg = this.$t('operations:vote.request', {origin: this.incoming.origin, entity: entity });
-                this.generictitle=this.$t('operations:vote.title');
-                this.genericaccept=this.$t('operations:vote.accept_btn');
-                this.genericreject=this.$t('operations:vote.reject_btn');
+                this.genericmsg = this.$t(
+                    'operations:vote.request',
+                    {
+                        origin: this.incoming.origin,
+                        entity: mappedData.entity
+                    }
+                );
+                this.generictitle = this.$t('operations:vote.title');
+                this.genericaccept = this.$t('operations:vote.accept_btn');
+                this.genericreject = this.$t('operations:vote.reject_btn');
```

Comment:
Much cleaner. Still, we need to think this part through better to work well for multiple chains. Probably need to standardize the form of the request data and implement a generic method to handle them here. Then each chain implements mapOperationData  for all API calls

Reply:
Yes, that is basically what I do to display the injectedCall.

My suggestion would be to adopt that way of formatting, and have a beautifier that can be applied optionally (chain dependent)